### ISSUE TITLE
Filtering Transition states: prophase to interphase--> remove prophase not interphase?

### ISSUE THREAD

**Issue by DeboraOlivier** [DeboraOlivier/NucleoTeloTrack#2]
Some interphase cells are misclassified as prophases & cycle between the states of interphase and prophase.

- if we remove the transition branch prophase to interphase, the algorithm removes all the interphase states following a misclassified prophase (instead of just removing the wrong prophase) ie the error is not the end point of the arrow but the beginning
it also leads to an accumulation of errors in the prophase class.

**Comment by DeboraOlivier** [comment]
examples: tracks after border exclusion: track 8 has lots of misclassified prophases
![siCTRL_20190524_Pos02_cl16_1border_0](https://user-images.githubusercontent.com/52743133/104921559-cb438400-5999-11eb-8924-37d91e05f457.png)

![siCTRL_20190524_Pos02_cl16_2rule_0](https://user-images.githubusercontent.com/52743133/104921562-cb438400-5999-11eb-8acc-95d504fedeb1.png)
track 8: rules exclusion removes all the interphases between prophases rather than removing the wrong prophase.

note: a correct prophase would proceed to prometaphase then meta etc

@msphan what do you think?

**Comment by msphan** [comment]
Thanks @DeboraOlivier. Unfortunately, the algorithm looks up only the current and the next states for instance to decide whether to exclude that next state. A quick solution but not optimized is to allow back transistion from prophase to interphase. Otherwise, I'm thinking add more constraints that check both previous and next states to reduce error propagation.

**Comment by DeboraOlivier** [comment]
Ok then let's leave it for now as "not urgent to do" (unless in the course
of your work you stumble on a quick solution).

**Comment by msphan** [comment]
I'll do it lately this week or next week :).

**Comment by msphan** [comment]
Updated remove_transistions() in trackprocessor.py on master branch. I introduced "super state" which is the group of identical states, e.g. the super states of the states [1,1,1,1,2,2,1,1,1] are [1,2,1]. Then, exclude the super state based on its previous and next states. 
<img width="932" alt="Capture d’écran 2021-03-01 à 23 09 09" src="https://user-images.githubusercontent.com/16255372/109565806-480e6580-7ae3-11eb-8288-af7b4100718a.png">
